File: 75. Sort Colors/75. Sort Colors.py
from typing import List


class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        low = 0
        curr = 0
        high = len(nums) - 1

        while curr <= high:
            if nums[curr] == 0:
                nums[low], nums[curr] = nums[curr], nums[low]
                low += 1
                curr += 1
            elif nums[curr] == 1:
                curr += 1
            else:
                nums[curr], nums[high] = nums[high], nums[curr]
                high -= 1

File: 75. Sort Colors/75. Sort Colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function (nums) {
  let low = 0;
  let curr = 0;
  let high = nums.length - 1;

  while (curr <= high) {
    if (nums[curr] === 0) {
      [nums[low], nums[curr]] = [nums[curr], nums[low]];
      low++;
      curr++;
    } else if (nums[curr] === 1) {
      curr++;
    } else {
      [nums[curr], nums[high]] = [nums[high], nums[curr]];
      high--;
    }
  }
};
